go/multipooler/manager/manager.go

```diff
@@ -345,36 +364,45 @@ func (pm *MultiPoolerManager) getPgCtldClient() pgctldpb.PgCtldClient {
 	return pm.pgctldClient
 }
 
-// getTableGroup returns the table group from the multipooler record
-func (pm *MultiPoolerManager) getTableGroup() string {
-	pm.mu.Lock()
-	defer pm.mu.Unlock()
-	if pm.multipooler != nil && pm.multipooler.MultiPooler != nil {
-		return pm.multipooler.TableGroup
+// getCachedTableGroup returns the table group from the multipooler record
```

Comment:
I wonder whether we should keep the original names for these functions. Putting `Cached` in the name makes it very explicit that we aren't touching the "real" object, but the fact that it is cached should be an internal implementation detail that is not exposed to callers. As far as callers are concerned, they ask MultiPoolerManager for its TableGroup / Shard / PoolerType and they get whatever it gives them.
Eventually we might also have to make these functions exported, because not all manager code will live in the manager package. But that is something we can do when the time comes.